### PR TITLE
[aws-json] Migrate SNS ListSubscriptions

### PR DESCRIPTION
**Pull request by kojisaikiAtSony**

### CODE REVIEW

**Comment by kojisaikiAtSony** [comment]
Hi @Admiral-Piett can you please review this?

File: smoke_tests/sns_list_subscriptions_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	assert.NotNil(t, subscribeResponse)
+	assert.NotNil(t, subscribeResponse2)
+
+	app.SyncTopics.Lock()
+	defer app.SyncTopics.Unlock()
+
+	// check listed subscriptions
+	sdkResponse, err := snsClient.ListSubscriptions(context.TODO(), &sns.ListSubscriptionsInput{})
+	for _, sub := range sdkResponse.Subscriptions {
+		fmt.Println(*sub.TopicArn)
```
Comment (Admiral-Piett) [inline smoke_tests/sns_list_subscriptions_test.go:98]: Do we need this print, or even this loop for that matter?

**Review by Admiral-Piett — COMMENTED** [review]
One tiny thing, otherwise looks great!

File: smoke_tests/sns_list_subscriptions_test.go (hunk 1 above)
Comment (kojisaikiAtSony) [reply smoke_tests/sns_list_subscriptions_test.go]: It wasn't required, unfortunately. Removed!

**Review by Admiral-Piett — APPROVED** [review]
Lovely, thank you!

**Pull request merged**